src/server/bg_services/replication_log_parser.js

```diff
@@ -0,0 +1,320 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const AWS = require('aws-sdk');
+const system_store = require('../system_services/system_store').get_instance();
+const replication_store = require('../system_services/replication_store').instance();
+const cloud_utils = require('../../util/cloud_utils');
+const http_utils = require('../../util/http_utils');
+const pool_server = require('../system_services/pool_server');
+const dbg = require('../../util/debug_module')(__filename);
+const config = require('../../../config');
+const moment = require('moment');
+
+/**
+ * get_log_candidates will return an object which contains the log candidates
+ * @param {*} source_bucket_id ID of the source bucket
+ * @param {String} rule_id ID of the replication rule
+ * @param {Record<any, any>} replicationconfig Replication configuration
+ * @param {Number} candidates_limit Candidates limit
+ * @param {(log_entry: Record<string, any>) => Boolean} ignore_fn - function that returns true if the log entry should be ignored
```

Comment:
In all other cases in this PR it is string (not `String`)

Reply:
It is `string`, isn't that correct?

Reply:
From the other examples in this PR, it seems `string`.
I also found [this](https://jsdoc.app/tags-param.html) and inside in the examples, it is `string`
![Screenshot 2023-02-27 at 13 34 28](https://user-images.githubusercontent.com/57721533/221553667-e8edca6c-d934-4476-be19-8a13c9a6dcb1.png)

Reply:
Oh no, I meant that the line that you have pointed to uses `string`.

Also, both `String` and `string` are valid types to use (one being the class another primitive type) in JSDoc. But I agree, maybe we can just use `string` instead of mixing `String` and `string`.

Reply:
Marking resolved - addressed.